src/scicookie/{{cookiecutter.project_slug}}/build-system/hatch-pyproject.toml

```diff
@@ -0,0 +1,119 @@
+[build-system]
+requires = ["hatchling>=1.17.1"]
+build-backend = "hatchling.build"
+
+[project]
+name = "{{ cookiecutter.project_name }}"
+authors = [
+  { name = "{{ cookiecutter.author_full_name }}", email = "{{ cookiecutter.author_email }}" },
+]
+description = "{{ cookiecutter.project_short_description }}"
+readme = "README.md"
+classifiers = [
+  "Development Status :: 1 - Planning",
+  "Intended Audience :: Science/Research",
+  "Intended Audience :: Developers",
+{%- if cookiecutter.project_license == "MIT" %}
+  "License :: OSI Approved :: MIT License",
+{%- elif cookiecutter.project_license == "BSD 3 Clause" %}
+  "License :: OSI Approved :: BSD License",
+{%- elif cookiecutter.project_license == "Apache Software License 2.0" %}
+  "License :: OSI Approved :: Apache Software License",
+{%- elif cookiecutter.project_license == "GNU General Public License v3" %}
+  "License :: OSI Approved :: GNU General Public License v3 (GPLv3)",
+{%- endif %}
+  "Operating System :: OS Independent",
+  "Programming Language :: Python",
+  "Programming Language :: Python :: 3",
+  "Programming Language :: Python :: 3 :: Only",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+  "Topic :: Scientific/Engineering",
+  "Typing :: Typed",
+]
+dynamic = ["version"]
```

Comment:
This probably won't fix the TOML error, but you must specify the file from which the version has to be read if you are setting it up dynamically - https://hatch.pypa.io/latest/version/#configuration -
```toml
[tool.hatch.version]
path = "path/to/version/file.py"
```

Reply:
`hatch` requires having either `version = "x.x.x"` or `__version__ = "x.x.x"` in the version file (in our case, `__init__.py`) (without the type signature (`: src`). We can update the regex to adapt to this project. If that does not work then we can have a condition in `__init__.py` to check for `hatch` and change the way `version` and `__version__` are declared.

Reply:
Hi, I have a question here.  In the __init__.py we don't have directly the version value but by the `get_version` function. How should this be configured then?

Reply:
I think the default regex that hatch uses to find the version string does not consider the typing annotation. It just looks for the variable `version` or `__version__` which is assigned a string without any typing annotation - https://hatch.pypa.io/latest/version/#configuration.

Also, given that hatch uses regex to find the version defined in a file, using a function (`get_version`) might not work as the function will never be executed.

Note how `scientific-python/cookie` also defines version just as a string without any type annotation -

https://github.com/scientific-python/cookie/blob/60cc884598c5abd8d300c293c528ebf0ffab2902/%7B%7Bcookiecutter.project_name%7D%7D/src/%7B%7Bcookiecutter.__project_slug%7D%7D/__init__.py#L13

Then they link that file in pyproject.toml, similar to the suggestion I made (and I think has been added in this PR) -

https://github.com/scientific-python/cookie/blob/60cc884598c5abd8d300c293c528ebf0ffab2902/%7B%7Bcookiecutter.project_name%7D%7D/pyproject.toml#L206

@ayeankit regarding the commit messages again, could you add some relevant messages to the commits that you are creating so we know what all you are trying while debugging? This would make reviewing this a bit easier too.